File: app/src/main/java/com/example/inventoryobat/ApiService.java
package com.example.inventoryobat;

import java.util.List;

import okhttp3.MultipartBody;
import okhttp3.RequestBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Part;
import retrofit2.http.Path;

public interface ApiService {

    // ========== OBAT ENDPOINTS ==========
    @GET("v1/obats")
    Call<ApiResponse<List<Obat>>> getAllObats();

    @GET("v1/obats/{id}")
    Call<ApiResponse<Obat>> getObatById(@Path("id") int id);

    @GET("v1/obats/jenis/{jenis}")
    Call<ApiResponse<List<Obat>>> getObatByJenis(@Path("jenis") String jenis);

    @Multipart
    @POST("v1/obats")
    Call<ApiResponse<Obat>> createObat(
            @Part("nama_obat") RequestBody nama,
            @Part("jenis_obat") RequestBody jenis,
            @Part("stock") RequestBody stock,
            @Part("id_supplier") RequestBody idSupplier,
            @Part MultipartBody.Part gambar
    );

    @Multipart
    @POST("v1/obats/{id}")
    Call<ApiResponse<Obat>> updateObatWithImage(
            @Path("id") int id,
            @Part("nama_obat") RequestBody nama,
            @Part("jenis_obat") RequestBody jenis,
            @Part("stock") RequestBody stock,
            @Part("id_supplier") RequestBody idSupplier,
            @Part MultipartBody.Part gambar,
            @Part("_method") RequestBody method
    );

    @FormUrlEncoded
    @PUT("v1/obats/{id}")
    Call<ApiResponse<Obat>> updateObatNoImage(
            @Path("id") int id,
            @Field("nama_obat") String nama,
            @Field("jenis_obat") String jenis,
            @Field("stock") int stock,
            @Field("id_supplier") int idSupplier
    );

    @PUT("v1/obats/{id}/stock")
    Call<ApiResponse<Obat>> updateStock(
            @Path("id") int id,
            @Body UpdateStockRequest request
    );

    @DELETE("v1/obats/{id}")
    Call<ApiResponse<String>> deleteObat(@Path("id") int id);

    // ========== SUPPLIER ENDPOINTS ==========
    @GET("v1/suppliers")
    Call<ApiResponse<List<Supplier>>> getAllSuppliers();

    @GET("v1/suppliers/{id}")
    Call<ApiResponse<Supplier>> getSupplierById(@Path("id") int id);

    @POST("v1/suppliers")
    Call<ApiResponse<Supplier>> createSupplier(@Body Supplier supplier);

    @PUT("v1/suppliers/{id}")
    Call<ApiResponse<Supplier>> updateSupplier(@Path("id") int id, @Body Supplier supplier);

    @DELETE("v1/suppliers/{id}")
    Call<ApiResponse<String>> deleteSupplier(@Path("id") int id);
}

File: app/src/main/java/com/example/inventoryobat/ApiService.kt
package com.example.inventoryobat

import okhttp3.MultipartBody
import okhttp3.RequestBody
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.GET
import retrofit2.http.Multipart
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Part
import retrofit2.http.Path

interface ApiService {

    @GET("v1/obats")
    fun getAllObats(): Call<ApiResponse<List<Obat>>>

    @GET("v1/obats/{id}")
    fun getObatById(@Path("id") id: Int): Call<ApiResponse<Obat>>

    @GET("v1/obats/jenis/{jenis}")
    fun getObatByJenis(@Path("jenis") jenis: String): Call<ApiResponse<List<Obat>>>

    @Multipart
    @POST("v1/obats")
    fun createObat(
        @Part("nama_obat") nama: RequestBody,
        @Part("jenis_obat") jenis: RequestBody,
        @Part("stock") stock: RequestBody,
        @Part("id_supplier") idSupplier: RequestBody,
        @Part gambar: MultipartBody.Part
    ): Call<ApiResponse<Obat>>

    @Multipart
    @POST("v1/obats/{id}")
    fun updateObatWithImage(
        @Path("id") id: Int,
        @Part("nama_obat") nama: RequestBody,
        @Part("jenis_obat") jenis: RequestBody,
        @Part("stock") stock: RequestBody,
        @Part("id_supplier") idSupplier: RequestBody,
        @Part gambar: MultipartBody.Part,
        @Part("_method") method: RequestBody
    ): Call<ApiResponse<Obat>>

    @FormUrlEncoded
    @PUT("v1/obats/{id}")
    fun updateObatNoImage(
        @Path("id") id: Int,
        @Field("nama_obat") nama: String,
        @Field("jenis_obat") jenis: String,
        @Field("stock") stock: Int,
        @Field("id_supplier") idSupplier: Int
    ): Call<ApiResponse<Obat>>

    @PUT("v1/obats/{id}/stock")
    fun updateStock(
        @Path("id") id: Int,
        @Body request: UpdateStockRequest
    ): Call<ApiResponse<Obat>>

    @DELETE("v1/obats/{id}")
    fun deleteObat(@Path("id") id: Int): Call<ApiResponse<String>>

    @GET("v1/suppliers")
    fun getAllSuppliers(): Call<ApiResponse<List<Supplier>>>

    @GET("v1/suppliers/{id}")
    fun getSupplierById(@Path("id") id: Int): Call<ApiResponse<Supplier>>

    @POST("v1/suppliers")
    fun createSupplier(@Body supplier: Supplier): Call<ApiResponse<Supplier>>

    @PUT("v1/suppliers/{id}")
    fun updateSupplier(@Path("id") id: Int, @Body supplier: Supplier): Call<ApiResponse<Supplier>>

    @DELETE("v1/suppliers/{id}")
    fun deleteSupplier(@Path("id") id: Int): Call<ApiResponse<String>>
}